include/mbedtls/x509_crt.h

```diff
@@ -841,6 +841,32 @@ typedef int (*mbedtls_x509_crt_ca_cb_t)( void *p_ctx,
                                          mbedtls_x509_crt const *child,
                                          mbedtls_x509_crt **candidate_cas );
 
+/**
+ * \brief               The type of trusted certificate CA indication callbacks.
+ *
+ *                      Callbacks of this type are passed to and used by the CRT
+ *                      verification routine mbedtls_x509_crt_verify_with_ca_cb()
+ *                      when looking for trusted signers of a given certificate.
+ *
+ *                      On success, the callback returns a list of trusted
+ *                      certificates to be considered as potential signers
+ *                      for the input certificate.
+ *
+ * \param p_ctx         An opaque context passed to the callback.
+ *                      This is the same opaque context passed to
+ *                      \p mbedtls_x509_crt_ca_cb_t and registered in
+ *                      \c mbedtls_ssl_conf_ca_cb()
+ * \param p             Buffer into which callback may write DN hints, with
+ *                      zero or more hints formatted as sequence of 2-byte
+ *                      big-endian length of DN followed by DER-encoded DN.
+ * \param len           Maximum length of DN hints that may be added to p.
+ *
+ * \return              Length in bytes of DistinguishedName list added.
+ */
+typedef size_t (*mbedtls_x509_crt_ca_hint_cb_t)( void *p_ctx,
```

Comment:
This does not belong in `x509_crt.h` but in `ssl.h`, as this only makes sense in TLS.

Reply:
:rocket: (removed the commit containing CA hint CB)